model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/clientscope/entity/JpaClientScopeMetadata.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.storage.jpa.clientscope.entity;
+
+import java.io.Serializable;
+import org.keycloak.models.map.clientscope.MapClientScopeEntityImpl;
+import org.keycloak.models.map.common.DeepCloner;
+
+public class JpaClientScopeMetadata extends MapClientScopeEntityImpl implements Serializable {
```

Comment:
Just to double check: this entityVersion tha we include in every metadata class can't be moved into the corresponding Map*Impl because it is JPA specific, right?

Reply:
I believe it's not just jpa specific. Each implementation should maintain an [_entity schema version_](https://github.com/keycloak/keycloak-community/blob/main/design/keycloak.x/storage-persistence.md#-version-compatibility-vc). 

IMO it requires wider discussion.

Reply:
It is specific to a particular implementation. Several implementations could share the same mechanism of versioning entities - then they could leverage the same entity schema version. This however cannot be applied generally to _every_ implementations, thus it cannot be moved into any of the generic map classes (including `Map*Impl`)

Reply:
Ok, makes sense. If we decide to refactor the Jpa*Metadata classes in the future we can do that as part of a different PR. Resolved, thanks!